perfkitbenchmarker/benchmarks/scimark2_benchmark.py

```diff
@@ -0,0 +1,265 @@
+        required to run the benchmark.
+
+  Returns:
+    A list of sample.Sample objects.
+  """
+  vms = benchmark_spec.vms
+  vm = vms[0]
+  logging.info('Running SciMark2 on %s', vm)
+  samples = []
+  if TEST_PARSE_RESULTS:
+    samples.extend(TestParseResults())
+    return samples
+  cmds = [
+      '(cd {0} && make c)'.format(SCIMARK2_PATH),
+      '(cd {0} && make java)'.format(SCIMARK2_PATH)
+  ]
+  for cmd in cmds:
+    stdout, _ = vm.RemoteCommand(cmd, should_log=True)
+    samples.extend(ParseResults(stdout))
+  return samples
+
+
+def Cleanup(benchmark_spec):
+  """Cleanup SciMark2 on the target vm.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+  """
+  pass
+
+
+def ParseResults(results):
+  """Result parser for SciMark2.
+
+  Sample Results (C version):
+    **                                                              **
+    ** SciMark2 Numeric Benchmark, see http://math.nist.gov/scimark **
+    ** for details. (Results can be submitted to pozo@nist.gov)     **
+    **                                                              **
+    Using       2.00 seconds min time per kenel.
+    Composite Score:         1596.04
+    FFT             Mflops:  1568.64    (N=1024)
+    SOR             Mflops:  1039.98    (100 x 100)
+    MonteCarlo:     Mflops:   497.64
+    Sparse matmult  Mflops:  1974.39    (N=1000, nz=5000)
+    LU              Mflops:  2899.56    (M=100, N=100)
+
+  Sample Results (Java version):
+
+    SciMark 2.0a
+
+    Composite Score: 1731.4467627163242
+    FFT (1024): 996.9938397943672
+    SOR (100x100):   1333.5328291027124
+    Monte Carlo : 724.5221517116782
+    Sparse matmult (N=1000, nz=5000): 1488.18620413327
+    LU (100x100): 4113.998788839592
+
+    java.vendor: Oracle Corporation
+    java.version: 1.7.0_75
+    os.arch: amd64
+    os.name: Linux
+    os.version: 3.16.0-25-generic
+
+  Args:
+    results: SciMark2 result.
+
+  Returns:
+    A list of sample.Sample objects.
+  """
+  RESULT_START_C = '** SciMark2 Numeric Benchmark'
+  RESULT_START_JAVA = 'SciMark 2.0a'
+
+  SCORE_REGEX = r'\n(Composite Score):\s+(\d+\.\d+)'
+  RESULT_REGEX_C = r'\n(.+?)\s+Mflops:\s+(\d+\.\d+)(\s+\(.+?\))?'
+  RESULT_REGEX_JAVA = r'\n([^:]+):\s+(\d+\.\d+)'
+  PLATFORM_REGEX = r'\n(\w+\.\w+):\s+(.*)'
+
+  def FindBenchStart(results, start_index=0):
+    bench_version = 'Unknown'
+    index = results.find(RESULT_START_C, start_index)
+    if index != -1:
+      bench_version = 'C'
+    else:
+      index = results.find(RESULT_START_JAVA, start_index)
+      if index != -1:
+        bench_version = 'Java'
+    return index, bench_version
+
+  def ExtractPlatform(result, bench_version, clear=True):
+    metadata = dict()
+    if bench_version == 'C':
+      pass
+    elif bench_version == 'Java':
+      matches = re.finditer(PLATFORM_REGEX, result)
+      for m in matches:
+        metadata.update({m.group(1): m.group(2)})
+        if clear:
```

Comment:
Is the "clear" functionality necessary for getting results, or is this primarily a debugging convenience? If it's not essential, I think it would be cleaner to remove it.

Style note: prefer putting parentheses around a long statement instead of using `\` line continuations:
    result = (
        stuff +
        more stuff +
        ...
    )